exonum/src/runtime/mod.rs

```diff
@@ -388,63 +400,61 @@ pub trait Runtime: Send + fmt::Debug + 'static {
-    /// (i.e., services that will be instantiated within the block do **not** receive a call)
-    /// exactly once for each block. The method is not called for the genesis block.
+    /// exactly once for each block. Services that will be instantiated within the block do **not**
+    /// receive a call. The method is not called for the genesis block.
     ///
-    /// # Return value
+    /// # Return Value
     ///
-    /// An error or panic returned from this method will lead to the rollback of all changes
-    /// in the fork enclosed in the `context`. Runtimes can, but are not required to convert panics
-    /// into errors.
+    /// An error returned from this method will lead to the rollback of all changes
+    /// in the fork enclosed in the `context`.
     fn before_transactions(
         &self,
         context: ExecutionContext<'_>,
         instance_id: InstanceId,
     ) -> Result<(), ExecutionError>;
 
-    /// Notifies a service stored in this runtime about the end of the block, allowing it
+    /// Notifies a service stored in this runtime about the end of the block. Allows the method
     /// to modify the blockchain state after all transactions in the block are processed.
     ///
     /// `after_transactions` is called for every service active at the beginning of the block
-    /// (i.e., services instantiated within the block do **not** receive a call) exactly
-    /// once for each block.
+    /// exactly once for each block. Services instantiated within the block do **not** receive a call.
     ///
     /// # Return value
     ///
-    /// An error or panic returned from this method will lead to the rollback of all changes
+    /// An error returned from this method will lead to the rollback of all changes
     /// in the fork enclosed in the `context`.
     fn after_transactions(
         &self,
         context: ExecutionContext<'_>,
         instance_id: InstanceId,
     ) -> Result<(), ExecutionError>;
 
-    /// Notifies the runtime about commitment of a new block.
+    /// Notifies the runtime about commit of a new block.
     ///
-    /// This method is guaranteed to be called *after* all `commit_service` calls related
+    /// This method is called *after* all `commit_service` calls related
     /// to the same block. The method is called exactly once for each block in the blockchain,
     /// including the genesis block.
     ///
-    /// A block is not yet persisted when this method is called; the up to date block information
-    /// is provided in the `snapshot`. It corresponds exactly to the information
-    /// eventually persisted; i.e., no modifying operations are performed on the block.
+    /// A block is not yet persisted when this method is called. The `snapshot` provides an up-to-date
+    /// block information. It corresponds exactly to the information
+    /// eventually persisted. This means that no modifying operations are performed on the block.
```

Comment:
Nit: Since the last sentence rewords the penultimate one, it could be safely removed.